lib/fileset/default.nix

```diff
@@ -709,4 +710,31 @@ in {
       throw "lib.fileset.gitTrackedWith: Expected the second argument (${toString path}) to point to a local working tree of a Git repository, but it's not."
     else
       _mirrorStorePath path fetchResult.outPath;
+
+  /*
+    Create a file set from a path that may or may not exist:
+    - If the path does exist, the path is [coerced to a file set](#sec-fileset-path-coercion).
+    - If the path does not exist, a file set containing no files is returned.
+
+    Type:
+      maybeMissing :: Path -> FileSet
+
+    Example:
+      # All files in the current directory, but excluding main.o if it exists
+      difference ./. (maybeMissing ./main.o)
+  */
+  maybeMissing =
+    path:
+    if ! isPath path then
+      if isStringLike path then
+        throw ''
+          lib.fileset.maybeMissing: Argument ("${toString path}") is a string-like value, but it should be a path instead.''
+      else
+        throw ''
+          lib.fileset.maybeMissing: Argument is of type ${typeOf path}, but it should be a path instead.''
```

Comment:
Same here.
Might be helpful to have a function that generates an explanation or something. Or just a let binding with the url?

Reply:
Just for the sake of consistency I'd prefer just copying it to all the error messages for now. We could consider having a dedicated `lib/fileset/errors.nix` in the future to collect and abstract over all error messages together.